sdk/storage/azdatalake/directory/client.go

```diff
@@ -279,9 +279,11 @@ func (d *Client) GetProperties(ctx context.Context, options *GetPropertiesOption
 	var respFromCtx *http.Response
 	ctxWithResp := shared.WithCaptureBlobResponse(ctx, &respFromCtx)
 	resp, err := d.blobClient().GetProperties(ctxWithResp, opts)
+	if err != nil {
+		return GetPropertiesResponse{}, exported.ConvertToDFSError(err)
```

Comment:
In case of error, we should return nil response instead of empty struct as response.